aloha/real_env.py

```diff
@@ -136,33 +127,56 @@ def setup_robots(self):
         setup_follower_bot(self.follower_bot_right)
 
     def get_qpos(self):
-        left_qpos_raw = self.recorder_left.qpos
-        right_qpos_raw = self.recorder_right.qpos
-        left_arm_qpos = left_qpos_raw[:6]
-        right_arm_qpos = right_qpos_raw[:6]
-        left_gripper_qpos = [FOLLOWER_GRIPPER_POSITION_NORMALIZE_FN(left_qpos_raw[7])]
-        right_gripper_qpos = [FOLLOWER_GRIPPER_POSITION_NORMALIZE_FN(right_qpos_raw[7])]
-        return np.concatenate(
-            [left_arm_qpos, left_gripper_qpos, right_arm_qpos, right_gripper_qpos]
-        )
+        # Initialize a list to hold the arm and gripper positions
+        qpos_list = []
+
+        # Iterate through all follower robots in the self.robots dictionary
+        for name, bot in self.robots.items():
+            if "follower" in name:
+                # Get the arm joint positions (first 6 joints)
+                arm_qpos = bot.core.joint_states.position[:6]
+                qpos_list.append(arm_qpos)
+
+                # Get the gripper joint position (7th joint) and normalize it
+                gripper_qpos = [FOLLOWER_GRIPPER_POSITION_NORMALIZE_FN(
+                    bot.core.joint_states.position[7])]
+                qpos_list.append(gripper_qpos)
+
+        # Concatenate all the positions into a single array
+        return np.concatenate(qpos_list)
 
     def get_qvel(self):
-        left_qvel_raw = self.recorder_left.qvel
-        right_qvel_raw = self.recorder_right.qvel
-        left_arm_qvel = left_qvel_raw[:6]
-        right_arm_qvel = right_qvel_raw[:6]
-        left_gripper_qvel = [FOLLOWER_GRIPPER_VELOCITY_NORMALIZE_FN(left_qvel_raw[7])]
-        right_gripper_qvel = [FOLLOWER_GRIPPER_VELOCITY_NORMALIZE_FN(right_qvel_raw[7])]
-        return np.concatenate(
-            [left_arm_qvel, left_gripper_qvel, right_arm_qvel, right_gripper_qvel]
-        )
+        # Initialize a list to hold the arm and gripper velocities
+        qvel_list = []
+
+        # Iterate through all follower robots in the self.robots dictionary
+        for name, bot in self.robots.items():
+            if "follower" in name:
+                # Get the arm joint velocities (first 6 joints)
+                arm_qvel = bot.core.joint_states.velocity[:6]
+                qvel_list.append(arm_qvel)
+
+                # Get the gripper joint velocity (7th joint) and normalize it
+                gripper_qvel = [FOLLOWER_GRIPPER_VELOCITY_NORMALIZE_FN(
+                    bot.core.joint_states.velocity[6])]
```

Comment:
We need method to get gripper velocity.

Reply:
https://github.com/Interbotix/interbotix_ros_toolboxes/pull/80/commits/d06005312c1db8cbb956ca05ecfb75f159fc19d4